File: SEP3_UserSystem/src/main/java/mediator/ServerConnector.java
package mediator;

import model.UserSystemModel;

import java.io.IOException;
import java.net.ServerSocket;
import java.net.Socket;

public class ServerConnector implements Runnable
{
    private UserSystemModel userSystemModel;
    private int PORT = 8888;
    private ServerSocket welcomeSocket;

    public ServerConnector(UserSystemModel userSystemModel)
    {
        this.userSystemModel = userSystemModel;
    }

    public void start() throws IOException
    {
        System.out.println("Starting UserSystem server...");
        welcomeSocket = new ServerSocket(PORT);

        while (true)
        {
            System.out.println("Wait for a client...");
            Socket socket = welcomeSocket.accept();
            System.out.println("Connect to a client...");
            ServerHandler server = new ServerHandler(userSystemModel,socket);
            Thread thread = new Thread(server);
            thread.setDaemon(true);
            thread.start();
        }
    }

    @Override public void run()
    {
        try
        {
            start();
        }
        catch (IOException e)
        {
            e.printStackTrace();
        }
    }
}

File: SEP3_ChatSystem/Mediator/ServerConnector.cs
using System;
using System.Net;
using System.Net.Sockets;
using System.Threading;
using SEP3_ChatSystem.Data;

namespace SEP3_ChatSystem.Mediator
{
    public class ServerConnector
    {
        private IChatModel chatModel;
        private string host;
        private int port;

        public ServerConnector(IChatModel chatModel,int port = 3010,string host = "127.0.0.1")
        {
            this.chatModel = chatModel;
            this.port = port;
            this.host = host;
        }

        public void Start()
        {
            Console.WriteLine("Starting ChatSystem server...");
            
            IPAddress ip = IPAddress.Parse(host);
            TcpListener listener = new TcpListener(ip, port);
            listener.Start();

            Console.WriteLine("Server started...");

            while (true)
            {
                Console.WriteLine("Wait for a client...");
                TcpClient client = listener.AcceptTcpClient();
                Console.WriteLine("Connect to a client...");
                new Thread(()=>new ServerHandler(chatModel,client)).Start();
            }
        }
    }
}